pubsub/azure/eventhubs/eventhubs.go

```diff
@@ -196,30 +514,42 @@ func (aeh *AzureEventHubs) Publish(req *pubsub.PublishRequest) error {
 
 // Subscribe receives data from Azure Event Hubs.
 func (aeh *AzureEventHubs) Subscribe(req pubsub.SubscribeRequest, handler pubsub.Handler) error {
-	cred, err := azblob.NewSharedKeyCredential(aeh.metadata.storageAccountName, aeh.metadata.storageAccountKey)
+	err := aeh.validateSubscriptionAttributes()
+	if err != nil {
+		return fmt.Errorf("error : error on subscribe %s", err)
+	}
+	if aeh.metadata.EnableEnitiyManagement {
+		if err = aeh.ensureSubscription(req.Topic); err != nil {
+			return err
+		}
+	}
+	cred, err := azblob.NewSharedKeyCredential(aeh.metadata.StorageAccountName, aeh.metadata.StorageAccountKey)
 	if err != nil {
 		return err
 	}
 
-	leaserCheckpointer, err := storage.NewStorageLeaserCheckpointer(cred, aeh.metadata.storageAccountName, aeh.metadata.storageContainerName, azure.PublicCloud)
+	// Set topic name, consumerID prefix for partition checkpoint lease blob path.
+	leaserPrefixOpt := storage.WithPrefixInBlobPath(aeh.getStoragePrefixString(req.Topic))
```

Comment:
This is needed to support multiple consumers with the same storage container option.